postgres/lib/dwconnect.js

```diff
@@ -738,11 +765,11 @@ module.exports = function (config, columnConfig) {
 				let joinTables = links.map(link => {
 					if (columnConfig.useSurrogateDateKeys && (link.table === 'd_datetime' || link.table === 'datetime' || link.table === 'dim_datetime')) {
 						sets.push(`${link.destination}_date = coalesce(t.${link.source}::date - '1400-01-01'::date + 10000, 1)`);
-						sets.push(`${link.destination}_time = coalesce(EXTRACT(EPOCH from t.${link.source}::time) + 10000, 1)`);
+						sets.push(`${link.destination}_time = coalesce(EXTRACT(EPOCH FROM ${(config.version !== 'redshift') ? `` : `'1970-01-01'::date +`} t.${link.source}::time) + 10000, 1)`);
```

Comment:
What does adding Jan 1, 1970 do here when extracting the epoch from the time?

Reply:
I can't see Michael's comments for some reason, but here's a small POC of Redshift acting different from Postgres:

;with dumb_behavior as (
    select top 10 _auditdate
    from d_order
    where _auditdate is not null
)
select _auditdate, extract(epoch from _auditdate::time), extract(epoch from '1/1/1970'::date + _auditdate::time)
from dumb_behavior;

The second column will return zeroes if run on Redshift.

Reply:
Sounds good.  I tested it with a hard coded timestamp and it worked.  Seems to work differently with how the column data is stored

Reply:
I wonder if the behavior is different between leader nodes and compute nodes.  Just Redshift things...